### ISSUE TITLE
Access to SampleLibrary size() - feature proposal

### ISSUE THREAD

**Issue by Barina** [godotengine/godot#5556]
**Operating system or device - Godot version:**
v2.0.3.stable.official

**Issue description** (what happened, and what was expected):
A feature proposal.
I wanted to know on runtime the size of a SampleLibrary content to play a random sample.
Therefore I can't find a way to get the size of samples (sound files) inside a SampleLibrary in script.
It'll be nice to have a 'size()' or a 'count()' or even a 'play_random()'

**Steps to reproduce:**

**Link to minimal example project** (optional but very welcome):

**Comment by J08nY** [comment]
Well there is `SampleLibrary.get_polyphony()` which returns `voices.size()` so I guess this is just a lack of documentation or do you require a different function?

**Comment by akien-mga** [comment]
`SampleLibrary.get_polyphony()` returns the number of voices/polyphony, i.e. how many samples can be played concurrently; as I understand it the feature request might be more for a way to know how many samples exist in the library in total. E.g. the polyphony could be 4 with a samples count of 8, which means you could only play 4 of the 8 samples at once.

**Comment by vnen** [comment]
BTW `get_polyphony()` is from `SamplePlayer` not from `SampleLibrary`.

**Comment by Barina** [comment]
Yes I meant what akien-mga mentioned

**Comment by J08nY** [comment]
Oh, I misread, sorry. In any case I'm on it.

**Comment by akien-mga** [comment]
What are your thought on what the new method should expose? Limiting is to the size of the library would be a bit sad IMO as it could instead give a complete list of samples in the library. The main question being whether it should be an array of `Sample`, or an array of `String` (names).

Since [most exposed methods rely on the name String](http://docs.godotengine.org/en/latest/classes/class_samplelibrary.html), I guess an array that return an array of String would be the most sensible, WDYT?

**Comment by J08nY** [comment]
Hmm yeah, I just did `SampleLibrary.size()` and `SampleLibrary.get_samples()` that returns and array of `StringName`s although now that I think about it, it really isn't a good name for the function. What about `SampleLibrary.get_sample_names()` ?

**Comment by akien-mga** [comment]
>  I just did `SampleLibrary.size()`

Which is the most common in the API so far, `get_size()` or `size()`?

>  `SampleLibrary.get_samples()` that returns and array of `StringName`s although now that I think about it, it really isn't a good name for the function. What about `SampleLibrary.get_sample_names()` ?

Yeah I think `get_sample_names()` is the best, as `get_samples()` might imply getting the `Sample` nodes.

**Comment by J08nY** [comment]
Well, 
```
: grep '<method name="size">' classes.xml | wc -l
9
: grep '<method name="get_size">' classes.xml | wc -l
0
```
Q.E.D. :smile:

**Comment by Sslaxx** [comment]
Sounds like we have an API inconsistency maybe. Should be get_size(), not just size()?

**Comment by akien-mga** [comment]
@Sslaxx See https://github.com/godotengine/godot/issues/5556#issuecomment-230522038 and https://github.com/godotengine/godot/pull/5561#issuecomment-230540406

**Comment by akien-mga** [comment]
Fixed by 49b288d, you can now use `get_sample_list.size()`.
